core/src/avm2/regexp.rs

```diff
@@ -144,9 +148,57 @@ impl<'gc> RegExp<'gc> {
         }
     }
 
-    pub fn exec(&mut self, text: AvmString<'gc>) -> Option<regress::Match> {
-        let global = self.flags.contains(RegExpFlags::GLOBAL);
-        let start = if global { self.last_index } else { 0 };
+    pub fn split(
+        &mut self,
+        activation: &mut Activation<'_, 'gc, '_>,
+        text: AvmString<'gc>,
+        limit: usize,
+    ) -> Result<Object<'gc>, Error> {
+        let mut storage = ArrayStorage::new(0);
+        // The empty regex is a special case which splits into characters.
+        if self.source.is_empty() {
+            let mut it = text.chars().take(limit);
+            while let Some(Ok(c)) = it.next() {
+                storage.push(
+                    AvmString::new(activation.context.gc_context, WString::from_char(c)).into(),
+                );
+            }
+            return ArrayObject::from_storage(activation, storage);
+        }
+
+        let mut start = 0;
+        let mut parts = 0;
```

Comment:
nit: `parts` can be replaced with `storage.len()`, which I see more elegant.

Reply:
Done. (also fixed some clippy findings)